extensions/dicom-sr/src/OHIFCornerstoneSRViewport.js

```diff
@@ -1,40 +1,63 @@
-import React, { useCallback, useEffect, useState } from 'react';
+import React, { useCallback, useContext, useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 import cornerstoneTools from 'cornerstone-tools';
 import cornerstone from 'cornerstone-core';
-
 import CornerstoneViewport from 'react-cornerstone-viewport';
-import OHIF, { utils } from '@ohif/core';
+import OHIF, { DicomMetadataStore, utils } from '@ohif/core';
 import { ViewportActionBar, useViewportGrid } from '@ohif/ui';
 import TOOL_NAMES from './constants/toolNames';
+import { adapters } from 'dcmjs';
+import getToolStateToCornerstoneMeasurementSchema from './utils/getToolStateToCornerstoneMeasurementSchema';
 import id from './id';
 
 const { formatDate } = utils;
 const scrollToIndex = cornerstoneTools.importInternal('util/scrollToIndex');
+const globalImageIdSpecificToolStateManager =
+  cornerstoneTools.globalImageIdSpecificToolStateManager;
 
-// const cine = viewportSpecificData.cine;
-
-// isPlaying = cine.isPlaying === true;
-// frameRate = cine.cineFrameRate || frameRate;
+const { StackManager, guid } = OHIF.utils;
 
-const { StackManager } = OHIF.utils;
+const MEASUREMENT_TRACKING_EXTENSION_ID = 'org.ohif.measurement-tracking';
 
 function OHIFCornerstoneSRViewport({
   children,
   dataSource,
   displaySet,
   viewportIndex,
-  DisplaySetService,
+  servicesManager,
+  extensionManager,
 }) {
+  const { DisplaySetService, MeasurementService } = servicesManager.services;
   const [viewportGrid, viewportGridService] = useViewportGrid();
   const [measurementSelected, setMeasurementSelected] = useState(0);
   const [measurementCount, setMeasurementCount] = useState(1);
   const [viewportData, setViewportData] = useState(null);
   const [activeDisplaySetData, setActiveDisplaySetData] = useState({});
   const [element, setElement] = useState(null);
-
+  const [isHydrated, setIsHydrated] = useState(displaySet.isHydrated);
```

Comment:
I think this is a good candidate for a JIRA Clarification issue. Should we only allow single hydration? (or expand current hydration workflow issue)